src/analysis/convex/cone.lean

```diff
@@ -643,4 +643,23 @@ begin
       set.mem_singleton_iff, forall_eq, imp_self] }
 end
 
+lemma closed_inner_dual_cone : is_closed (s.inner_dual_cone : set H) :=
+begin
+  -- reduce the problem to showing that dual cone of a singleton `{x}` is closed
+  rw inner_dual_cone_eq_Inter_inner_dual_cone_singleton,
+  apply is_closed_Inter,
+  simp_rw [set_coe.forall, subtype.coe_mk],
+  intros x hx,
+
+  -- the dual cone of a singleton set is the preimage of `[0, ∞)` under `inner x`
+  have h : (({x} : set H).inner_dual_cone : set H) = (inner x : H → ℝ)⁻¹' (set.Ici 0),
+  { apply set.ext,
```

Comment:
```suggestion
  { ext,
```
I haven't tried it though (gitpod did not start).

Reply:
I thought that `ext` is slower than `set.ext` as it has to run through all the extensionality arguments. It internally just does `set.ext` eventually.

Reply:
`ext` is not a tactic that is considered to be expensive, otherwise it would be totally useless (it almost always unfolds to something like `refine foo.ext (λ x, _)` .

Reply:
I see, thanks! I'll make this change.